tests/test_download_url_yandex.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import os
+import tempfile
+import unittest
+from urllib.error import HTTPError
+
+from monai.apps.utils import download_url
+
+YANDEX_MODEL_URL = (
+    "https://cloud-api.yandex.net/v1/disk/public/resources/download?"
+    "public_key=https%3A%2F%2Fdisk.yandex.ru%2Fd%2Fxs0gzlj2_irgWA"
```

Comment:
this test will run regularly by testing pipelines, do you think this URL and the file will be available for those? @blacky-i

Reply:
There are two risks in this url:
1. (API)[https://yandex.ru/dev/disk/api/reference/public.html#download] deprecation (which is very unlikely)
2. This little file is on my personal yandex disk, I will not delete it.

Reply:
ok, it seems when there are concurrent downloadings from multiple jobs, it becomes unavailable, for example:

https://github.com/Project-MONAI/MONAI/actions/runs/5403919298/jobs/9817414158

```
======================================================================
ERROR: test_verify (tests.test_download_url_yandex.TestDownloadUrlYandex)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/MONAI/MONAI/tests/test_download_url_yandex.py", line 34, in test_verify
    download_url(url=YANDEX_MODEL_URL, filepath=os.path.join(tempdir, "model.pt"))
  File "/home/runner/work/MONAI/MONAI/monai/apps/utils.py", line 208, in download_url
    with urlopen(url) as response:
  File "/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: NOT FOUND

----------------------------------------------------------------------
```